shiny_app/server.R

```diff
@@ -351,4 +351,153 @@ shinyServer(function(input, output, session) {
       })
     })
 
+
+  ## Surge Tabs ####
+
+  summary_outputs <- reactive({
+    model_output() %>%
+      filter(near(time, round(time))) %>%
+      group_by_at(vars(time:treatment)) %>%
+      summarise_all(sum) %>%
+      rename(treatment_pathway = treatment)
+  })
+
+  ## Tab - Subpopn ####
+
+  output$surge_subpopn <- renderTable({
+    summary_outputs() %>%
+      pivot_wider(names_from = type, values_from = value) %>%
+      dplyr::group_by(group) %>%
+      filter(!is.na(group)) %>%
+      summarise(`Adjusted exposed / at risk @ baseline` = as.integer(round(sum(`new-at-risk`, na.rm = T), 0)),
+                `Total symptomatic over period referrals` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+                `Total receiving services over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+      arrange(-`Total symptomatic over period referrals`)
+  })
+
+  ## Tab - Conditions ####
+
+  output$surge_condition <- renderTable({
+    summary_outputs() %>%
+      pivot_wider(names_from = type, values_from = value) %>%
+      dplyr::group_by(condition) %>%
+      filter(!is.na(condition)) %>%
+      summarise(`Adjusted exposed / at risk @ baseline` = as.integer(round(sum(`new-at-risk`, na.rm = T), 0)),
+                `Total symptomatic over period referrals` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+                `Total receiving services over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+      arrange(-`Total symptomatic over period referrals`)
+  })
+
+
+  ## Tab - Treatment Pathway ####
+
+  summary_treatment_pathway <- reactive({
+    summary_outputs() %>%
+    pivot_wider(names_from = type, values_from = value) %>%
+    dplyr::group_by(treatment_pathway) %>%
+    filter(!is.na(treatment_pathway)) %>%
+    summarise(`Total new referrals/presentations over period` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+              `Total services offered over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+    arrange(-`Total new referrals/presentations over period`)
+  })
+
+  output$surge_treatmentpathway <- renderTable({
+    summary_treatment_pathway()
+  })
+
+  output$surge_treatmentpathwayplot <- renderPlot({
+    summary_treatment_pathway() %>%
+      pivot_longer(!contains("treatment_pathway")) %>%
+      ggplot(aes(treatment_pathway, value)) +
+      geom_col(position = "dodge") +
+      labs(y = "Number",
+             x = "Treatment Pathway") +
+      scale_x_discrete(labels = function(x) str_wrap(x, 8),
+                       guide = guide_axis(n.dodge = 2)) +
+      facet_wrap(~ name, ncol=1)
```

Comment:
I would probably migrate to the more modern ggplot/tidy syntax of `facet_wrap(vars(name), ncol = 1)`